### ISSUE TITLE
[FAILED] Failed to run aws-provider-4.64.0 with openTofu

### ISSUE THREAD

**Issue by zhenrong-wang** [opentofu/opentofu#930]
This is a follow-up test, previous test with aliyun-provider, please check this issue: https://github.com/opentofu/opentofu/issues/929

I followed the same method and tested aws just now. It turns out the aws provider failed to start, and caused my Ubuntu system halt.

**Versions: openTofu (1.6.0 alpha5), terraform (1.6.2), aws-provider (4.64.0)**

**Results**

**A. [hpcopr](https://github.com/zhenrong-wang/hpc-now) with terraform:**

Creating a stack cost 123 seconds, while destroying it took 230 seconds. 

Somehow the trace log of terraform is very large (35 MB). So I zipped them, please check this out: 
[terraform-trace-creating-aws.zip](https://github.com/opentofu/opentofu/files/13482390/terraform-trace-creating-aws.zip)

**B. [hpcopr](https://github.com/zhenrong-wang/hpc-now) with openTofu:**

The provider started communicating with AWS API after 230 seconds of the `apply` command. And finally caused the Ubuntu system stop responding, and I had to restart the system by force (I used VMwarkstation, and the Ubuntu vm has 2 cores + 4 GB Memory). 

Check the screenshot below:

![屏幕截图 2023-11-28 092924](https://github.com/opentofu/opentofu/assets/59274212/1730909f-d47c-4c12-b87b-f726ac17ae3b)

After rebooting my system, I exported the failed-log, please check this out:

[tofu-trace-aws-creating-FAILED.zip](https://github.com/opentofu/opentofu/files/13482455/tofu-trace-aws-creating-FAILED.zip)

**Some interesting behaviors:**

a. There are too many provider processes, see the window at the right-bottom.
b. The trace log shows that the provider didn't work in a continuous way. It exited and restarted frequently during the `apply` process.

During the test, nothing but the tf binary changed. Because my HCL resource files were automatically-generated by the [hpcopr](https://github.com/zhenrong-wang/hpc-now) , so the target resource stack should be the same.

Could you please help on this issue? Thanks!

**Comment by zhenrong-wang** [comment]
Follow-up:

I also tried to run **tofu plan** command without [hpcopr](https://github.com/zhenrong-wang/hpc-now), and it turns out the provider and tofu failed to output plan to the terminal in the first 3 minutes.Check the screenshot below:

![屏幕截图 2023-11-28 095318](https://github.com/opentofu/opentofu/assets/59274212/93aff853-b318-4ab3-a26c-9f1e5bba856c)

After 4 minutes, it started communicating with AWS API to query some data. However, again, my Ubuntu vm stopped responding. Check the screen shot below:
![屏幕截图 2023-11-28 095712](https://github.com/opentofu/opentofu/assets/59274212/a69c7a95-a022-41b9-b577-29ac1377759a)

Finally, the provider seems exited. However, the Ubuntu stopped responding completely. Here is the final screen before I rebooting the os:
![image](https://github.com/opentofu/opentofu/assets/59274212/fe3a44b9-6f3a-4c98-8192-f2e550d0e1ce)

Given that it failed to output a plan for more than 4 minutes, I canceled the operation, and switched to terraform. 

All the HPC resource files kept unchanged. I successfully run the terraform init and terraform plan command in a quick way. Nothing abnormal happened. Please check the screenshot here:

![屏幕截图 2023-11-28 100649](https://github.com/opentofu/opentofu/assets/59274212/71f74e8f-9aa6-4445-9447-90df6d453e07)

I zipped 3 log files: terraform plan standard output, terraform-trace log, tofu-trace log. Please check the zipped file below.

[trace-plan-aws-test.zip](https://github.com/opentofu/opentofu/files/13482609/trace-plan-aws-test.zip)

**Comment by zhenrong-wang** [comment]
In summary, the [hpcopr](https://github.com/zhenrong-wang/hpc-now) wrapped tofu/terraform in a correct way. Running [hpcopr](https://github.com/zhenrong-wang/hpc-now) equals to running tofu/terraform.

Given that [hpcopr](https://github.com/zhenrong-wang/hpc-now) now support 7 different cloud services. It is quite convinient to use it to test openTofu and compare tofu with terraform easily.

I will test gcp, azurerm, azureaz, baidubce, tencentcloud, huaweicloud in the next few days.

Currently, the [hpcopr](https://github.com/zhenrong-wang/hpc-now) works with terraform and its providers well. I hope to switch smoothly from the **not-really-open-source** terraform to the **really-free-and-open-source** openTofu.

Thank you very much for your support!

**Comment by kislerdm** [comment]
Hey @zhenrong-wang! Thanks for followups.

Could you please share details about your setup: machine, OS, architecture; it'd be especially important to understand this:

> [...] my Ubuntu vm [...]

Could you please clarify if you used a VM on your personal Windows/Linux machine, or if was it an EC2 (or similar service) instance?

> After 4 minutes, it started communicating with AWS API

Could you please share the following: 
- Services in which AWS region did attempt to provision? 
- Have you tried to communicate to AWS using [awscli](https://aws.amazon.com/cli/) to eliminate network as a factor?

Thanks!

**Comment by kislerdm** [comment]
@zhenrong-wang also, please use the bug report [template](https://github.com/opentofu/opentofu/issues/new?assignees=&labels=bug%2Cpending-decision&projects=&template=bug_report.yml) and share the logs following the submission form instead of attaching screenshot.. Thanks!

**Comment by zhenrong-wang** [comment]
> @zhenrong-wang also, please use the bug report [template](https://github.com/opentofu/opentofu/issues/new?assignees=&labels=bug%2Cpending-decision&projects=&template=bug_report.yml) and share the logs following the submission form instead of attaching screenshots. Thanks!

Thanks for the reply.

Sure I'll follow the bug report guidelines and avoid screenshots.

From the logs, I think the network to AWS API was not the issue. Because the 2 tests (terraform vs tofu) were under the same network conditions.

Another words, I also tested azure providers, azurerm and azuread. The provider also took 150+ seconds before printing out the first line of stdout. And after the maximum time we set (600 seconds), the creating process didn't finish. Previously with terraform 1.6.2 or 1.5.0, it costs less than 300 seconds in all cases. That's why we set the maximum execution time of 600 seconds. 

If the logs are useful for you to debug, I'd be glad to test again and attach to this post.

Thanks a lot for your support!
